### PR TITLE
Hardwaregain and harmonic values test functions

### PR DESCRIPTION
**Pull request by ranechita**
# Description

Add test functions for checking that hardware gain is within certain intervals. Add test function for harmonic values, checking if the first few harmonics are in a given range.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How has this been tested?

**Test Configuration**:
* Hardware: ADRV9361-ZC702
* OS: ADI-Kuiper-Linux 2019R2 RC11

# Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have signed off all commits and they contain "Signed-off by: <insert name>"
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: test/dma_tests.py [hunk 1]
```diff
@@ -542,6 +543,121 @@ def gain_check(uri, classname, channel, param_set, dds_scale, min_rssi, max_rssi
     assert rssi >= min_rssi
     assert rssi <= max_rssi
 
+def hardwaregain(uri, classname, channel, dds_scale, frequency, hardwaregain_low, hardwaregain_high):
```
Comment (tfcollins) [inline test/dma_tests.py:546]: I don't really understand what this does. Does this measure changing in the AGC?

File: test/dma_tests.py [hunk 2]
```diff
@@ -542,6 +543,121 @@ def gain_check(uri, classname, channel, param_set, dds_scale, min_rssi, max_rssi
...
+    sdr.dds_single_tone(frequency, dds_scale, channel)
+    time.sleep(3)
+    
+    chname = "voltage" + str(channel)
+    hwgain = sdr._get_iio_attr(chname,"hardwaregain", False, sdr._ctrl)
+    print(hwgain)
+    assert hardwaregain_low <= hwgain <= hardwaregain_high
+
+def harmonic_vals(classname, uri, channel, param_set, low, high, plot=False):
+    """ harmonic_vals: Test first five harmonics and check to be within
```
Comment (tfcollins) [inline test/dma_tests.py:582]: The description needs an update. DMA is used not the DDSs

Comment (ranechita) [reply test/dma_tests.py:582]: I apologize, I updated it now

File: test/dma_tests.py (hunk 1 above)
Comment (ranechita) [reply test/dma_tests.py:546]: yes, i would like to use it for a series of tests for fmcomms2

Comment (tfcollins) [reply test/dma_tests.py]: Can you update the description a bit to explain its related to AGC? That way others will be able to leverage it more easily later on.

Comment (ranechita) [reply test/dma_tests.py]: Sorry for the delay, the description was updated. Is the current one appropriate?

**Pull request merged**